### PR TITLE
Plugin stepsform (wb-steps): Provides the ability to split a form into steps

### PR DESCRIPTION
**Pull request by ghost**
Currently the issue is long forms on the business side. This well make forms feel shorter for the customer/user. So far the plugin only converts form fields into steps.

Please see Issue #7898 and demo at kodz.ca/projects/wet-boew/demos/forms/steps

### CODE REVIEW

**Comment by kodecount** [comment]
Also a live example http://kodecount.ca/demos/formvalid/steps

**Comment by kodecount** [comment]
@nschonni @duboisp @LaurentGoderre Any tips?

File: src/plugins/stepsform/stepsform.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+							// continue if valid
+							if ( isFormValid ) {
+								this.parentElement.previousElementSibling.removeAttribute( "class" );
+								displayFieldset( this.parentElement, true );
+							}
+
+						} );
+						div.appendChild( next );
+					} else {
+						var value = ( settings && settings.submitBtnVal ) ? settings.submitBtnVal : "X";
```
Comment (kodecount) [inline src/plugins/stepsform/stepsform.js]: - [ ] i18n Text (add "Submit" as default button text (enabling tmpl-submit (not sure about this)))

File: src/plugins/stepsform/stepsform.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+								if ( isFormValid ) {
+									this.parentElement.previousElementSibling.removeAttribute( "class" );
+									displayFieldset( this.parentElement, true );
+								}
+
+							} );
+							div.appendChild( next );
+						}
+					} else {
+						var value = ( settings && settings.submitBtnVal ) ? settings.submitBtnVal : "X",
```
Comment (kodecount) [inline src/plugins/stepsform/stepsform.js]: - [ ] i18n Text (add "Submit" as default button text (enabling tmpl-submit (not sure about this)))

File: src/plugins/stepsform/stepsform.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+				};
+			}
+
+			// remove submit button (user should add one for no script purposes)
+			$( eventTarget ).find( "input[type=submit]" ).remove();
+
+			// find all fieldsets, hide required content, and add appropriate buttons
+			var fieldsets = eventTarget.getElementsByTagName( "FIELDSET" ),
+				settings = wb.getData( eventTarget, componentName ),
+				btnValue = ( settings && settings.submitBtnVal ) ? settings.submitBtnVal : "X",
```
Comment (kodecount) [inline src/plugins/stepsform/stepsform.js]: - [ ] i18n Text (add "Submit" as default button text (enabling tmpl-submit (not sure about this)))

**Comment by kodecount** [comment]
Travis failing, but it isn't related to my code (at least I don't think so).

File: src/plugins/stepsform/stepsform-fr.hbs [hunk 4]
```diff
@@ -0,0 +1,107 @@
+---
+{
+	"title": "Steps Form",
+	"language": "fr",
+	"category": "Plugins",
```
Comment (duboisp) [inline src/plugins/stepsform/stepsform-fr.hbs]: This should be "Plugiciels", that is why the Travis build are failing

Comment (kodecount) [inline src/plugins/stepsform/stepsform-fr.hbs]: Thanks.

File: src/plugins/stepsform/stepsform.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+					if ( !isLastFieldset ) {
+						btnClone = btnNext.cloneNode( true );
+						btnClone.id = componentName + "-btn-" + idCount;
+						idCount += 1;
+
+						setStepsBtnEvent( btnClone );
+						div.appendChild( btnClone );
+					} else {
+						var btnValue = ( settings && settings.submitBtnVal ) ? settings.submitBtnVal : "X",
```
Comment (kodecount) [inline src/plugins/stepsform/stepsform.js]: - [ ] i18n Text (add "Submit" as default button text (enabling tmpl-submit (not sure about this)))

Could also use `<i class="glyphicon glyphicon-send" aria-hidden="true"></i>` but accessibility would be an issue.

**Comment by kodecount** [comment]
As for the hitting submit/next twice bug their a few open/close issue:
https://github.com/jquery-validation/jquery-validation/issues/997
https://github.com/jquery-validation/jquery-validation/issues/1872

**Comment by kodecount** [comment]
Something like this at [Line 274](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/formvalid/formvalid.js#L274):
```javascript
/* fixes validation issue */
$form.on( "change", "input[type=date], input[type=file], select", function() {
  $form.validate().element( this );
} );
```

**Comment by kodecount** [comment]
Something like this at [Line 274](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/formvalid/formvalid.js#L274):
```javascript
/* fixes validation issue */
$form.on( "change", "input[type=date], input[type=file], select", function() {
  $form.validate().element( this );
} );
```
I would put this in a different PR if any think it be okay.

@RobJohnston This would address your issue with the files.

**Comment by duboisp** [comment]
As discussed at the PR review meeting, this PR would be a good candidate to be moved in the theme (GC web and intranet), with the other plugins that execute business logic.

**Comment by neoinsight** [comment]
@kodecount This looks really interesting :)  Could steps be implemented with field-flow or only form inputs? We are looking for ways to implement wizards with business logic @borges-porteous @patlaj

**Comment by kodecount** [comment]
I don't believe field-flow is currently part of WET (gcweb and intranet) but a project by @duboisp, but the plugin could be adapted after field-flow is part of WET (gcweb and intranet).

**Comment by lucas-hay** [comment]
@kodecount do you have any updates on this?
Will this be completed?
Field flow is now part of GCWEB (not WET), so perhaps maybe this should move over to GCWEB?

**Comment by kodecount** [comment]
I'm no longer with the project full time.

**Comment by duboisp** [comment]
It's now included in wet-boew as a provisional plugin via #8811

**Pull request closed without merging**